### PR TITLE
Issue 9 resolution

### LINKED ISSUE #9: [BUG]: Resolve ALL critical logic errors blocking system operation
# Logic Bugs: Critical Issues in Arduino/ESP32 code

**Multiple logic errors exist in the current implementation that must be addressed for the precision triangulation and landing system to function reliably.**

---
### 1. Global variable not updated (init hang)
- `maxKnown` flag in `src/arduino/comunication.cpp` shadows the global declaration.
- Arduino setup can hang forever.

### 2. Wrong I2C address usage
- ESP32 uses address `0x00` (reserved).
- Arduino/ESP32 may conflict and fail I2C comms.

### 3. Incorrect median filter for even data
- Median index computation is wrong for even number of valid samples.
- Outlier handling fails—erroneous coordinates result.

### 4. Trilateration math is incorrect
- Position calculation formula in `src/esp32/positioning.cpp` mixes up terms and coordinate references.
- Leads to invalid position output.

### 5. Type conversion/I2C data ordering
- Byte order mismatch and improper use of `Wire.read()` may result in corrupted distance values.

### 6. Sensor pin configuration
- Pin numbers (22-27) in `src/arduino/config.h` do not exist on Arduino Uno.
- No real sensor inputs are possible.

### 7. Uninitialized variables
- ESP32 global variables for coordinates, distances etc. are uninitialized when setup is skipped.
- Results in garbage/calculation errors.

### 8. Too slow sensor polling
- Excessive delay and timeout in `pulseIn()` causes system slowdowns and timeouts.

---
## Impact

- System hangs, invalid calculations, hardware setup failure, communication breakdown.
- Unable to test or operate safely—risk of drone loss or crash.

---
## What to Do

- Fix all logic bugs above and test edge cases.
- Review every global, initialization, pin mapping, comms protocol, and geometry.
- Provide proof of correctness for median and position calculations.
- Update documentation for pinouts, sensor behavior, and I2C protocol.

---

### PR DESCRIPTION
**Pull request by TheCipo**
Fix critical logic errors in Arduino/ESP32 core systems: variable shadowing, I2C addressing, trilateration math, type conversion, pin configuration, uninitialized variables, and sensor polling delays.

### CODE REVIEW

File: src/arduino/comunication.cpp [hunk 1]
```diff
@@ -5,10 +5,13 @@
 void I2C_DataSend() {
   Serial.print(MYADDRESS);
   byte buffer[6];
+
   for (int i = 0; i < 3; i++) {
-    buffer[i * 2] = lowByte(toSendDistances[i]);  // Byte basso
-    buffer[i * 2 + 1] = highByte(toSendDistances[i]); // Byte alto
+    int16_t value = (int16_t)toSendDistances[i];
+    buffer[i * 2] = (uint8_t)(value & 0xFF);
```
[the remark this answers is not in the document]
Comment (TheCipo) [reply src/arduino/comunication.cpp:11]: serial log necessary for debugging
